cli/tests/NetworkIO_unittest.cpp

```diff
@@ -0,0 +1,182 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ **/
+
+#include <cstddef>
+#include <cstdio>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "cli/Flags.hpp"
+#include "cli/LineReaderBuffered.hpp"
+#include "cli/NetworkCliClient.hpp"
+#include "cli/NetworkIO.hpp"
+#include "threading/ConditionVariable.hpp"
+#include "threading/Mutex.hpp"
+#include "threading/Thread.hpp"
+
+#include "glog/logging.h"
+#include "gtest/gtest.h"
+
+using std::unique_ptr;
+
+namespace quickstep {
+
+static std::string const kQueryRequest = "O Captain! My Captain!";
+static std::string const kQueryResponse = "Our fearful trip is done,";
+
+/**
+ * Contains a server instance for testing.
+ */
+class TestNetworkIO {
+ public:
+  TestNetworkIO()
+      : service_(),
+        server_address_("localhost:" + std::to_string(FLAGS_port)),
+        server_(nullptr) {
+    grpc::ServerBuilder builder;
+    builder.AddListeningPort(server_address_, grpc::InsecureServerCredentials());
+    builder.RegisterService(&service_);
+    server_ = builder.BuildAndStart();
+    CHECK(server_) << "Failed to start server";
+    LOG(INFO) << "TestSingleNodeServer listening on " << server_address_;
+  }
+
+  // Gets a message from the Service worker.
+  std::string getSentMessage() {
+    networkio_internal::RequestState& requestState = service_.getRequestState();
```

Comment:
Change to `networkio_internal::RequestState &request_state`.

Ditto in `setResponse`.